unresolved external symbol stx::v1::begin_panic

**Describe the bug**
When building the library with STX_BUILD_SHARED on MSVC (VS2019 16.7.3), trying to use it leads to a linker error:
```
error LNK2019: unresolved external symbol "void __cdecl stx::v1::begin_panic(
    class std::basic_string_view<char,struct std::char_traits<char> > const &,
    class stx::v1::ReportPayload const &,
    struct stx::v1::SourceLocation const &)" 
(?begin_panic@v1@stx@@YAXAEBV?$basic_string_view@DU?$char_traits@D@std@@@std@@AEBVReportPayload@12@AEBUSourceLocation@12@@Z)
referenced in function "void __cdecl stx::v1::panic<struct io::Error>(
    class std::basic_string_view<char,struct std::char_traits<char> > const &,
    struct io::Error const &,
    struct stx::v1::SourceLocation const &)" 
(??$panic@UError@io@@@v1@stx@@YAXAEBV?$basic_string_view@DU?$char_traits@D@std@@@std@@AEBUError@io@@AEBUSourceLocation@01@@Z)
```

The `begin_panic` function is called from the `panic` function template, which gets instantiated in user-code, 
so I think `begin_panic` would have to be conditionally marked STX_EXPORT as well.

**Expected behavior**
If using the library as a DLL on windows is a supported scenario, it should work.

**Desktop:**
 - OS: Windows 10
 - Version 19041

--- a/include/stx/panic.h
+++ b/include/stx/panic.h
@@ -48,7 +48,13 @@ void panic_handler(std::string_view const& info, ReportPayload const& payload,
 ///
 /// DO NOT INVOKE THIS FUNCTION!!!
 ///
-[[noreturn]] void begin_panic(std::string_view const& info,
+[[noreturn]]
+#if defined(STX_VISIBLE_PANIC_HOOK)
+STX_EXPORT
+#else
+STX_LOCAL
+#endif
+void begin_panic(std::string_view const& info,
                               ReportPayload const& payload,
                               SourceLocation const& location) noexcept;
 

--- a/src/panic/hook.cc
+++ b/src/panic/hook.cc
@@ -92,7 +92,13 @@ bool take_panic_hook(PanicHook* out) noexcept {
   return true;
 }
 
-[[noreturn]] void begin_panic(std::string_view const& info,
+[[noreturn]]
+#if defined(STX_VISIBLE_PANIC_HOOK)
+STX_EXPORT
+#else
+STX_LOCAL
+#endif
+void begin_panic(std::string_view const& info,
                               ReportPayload const& payload,
                               SourceLocation const& location) noexcept {
   // TODO(lamarrr): We probably need a method for stack unwinding, So we can

